src/main/java/games/strategy/engine/data/export/GameDataExporter.java

```diff
@@ -651,7 +652,7 @@ private Connection(final Territory t1, final Territory t2) {
 
     @Override
     public int hashCode() {
-      return territory1.hashCode() + territory2.hashCode();
+      return Objects.hash(territory1, territory2);
```

Comment:
Similar to what we discussed in `RelationshipTracker`.  There's no Javadoc here, but the semantics appear to be the same (unless `Connection` objects imply a direction).  Please consider partially reverting the implementation so that it produces the same hash code if `territory1` and `territory2` are swapped.